automated MM-14632 Integration test for removing team icon #4066

VishalSwarnkar opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->

#### Ticket Link (https://github.com/mattermost/mattermost-server/issues/10470)
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/10470

Otherwise, link the JIRA ticket.
-->

#### Related Pull Requests
<!--
List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
-->
- Has webapp changes (https://github.com/mattermost/mattermost-webapp/pull/4611/commits/05fb09b4eb642205ef5691280721a2b39814e1be)

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.
-->

mattermod commented:
Thanks @VishalSwarnkar for the pull request!

 Per the [**CONTRIBUTING.md**](https://github.com/mattermost/platform/blob/master/CONTRIBUTING.md#submitting-a-pull-request) file displayed when you created this pull request, we need to add you to the list of [approved contributors](https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true) for the Mattermost project.

**Please help complete the Mattermost [contribution license agreement](http://www.mattermost.org/mattermost-contributor-agreement/)?**

This is a standard procedure for many open source projects. Your form should be processed within 24 hours and reviewers for your pull request will be able to proceed.

Please let us know if you have any questions.

We are very happy to have you join our growing community! If you're not yet a member, please consider joining our [Contributors community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors and discuss new opportunities with the core team.

VishalSwarnkar commented:
@jasonblais, kindly review my PR, please.

jasonblais commented:
Thanks @VishalSwarnkar! Would you be open to signing the above contributor license agreement, standard in open source projects? If you have any questions, don't hesitate to reach out.

Once signed, we'll queue your PR for reviews. Big thanks for your contribution!

VishalSwarnkar commented:
> Thanks @VishalSwarnkar! Would you be open to signing the above contributor license agreement, standard in open source projects? If you have any questions, don't hesitate to reach out.
> 
> Once signed, we'll queue your PR for reviews. Big thanks for your contribution!

VishalSwarnkar commented:
Thanks, @jasonblais I have signed and submitted CLA

jasonblais commented:
Thanks @VishalSwarnkar, I've queued it for review. Would you be open to updating the PR description with a summary and ticket link?

VishalSwarnkar commented:
Thanks, @reflog for reviewing the code. As suggested I have replaced newly introduced _id_ with _data-testid_ and updated the PR. Kindly review the changes

jasonblais commented:
/check-cla

VishalSwarnkar commented:
Hi @jasonblais the PR is now approved now but still, I can see the current status as "merge/blocked Expected — Waiting for status to be reported". Is there anything I have to do in regards to the PR?

saturninoabril commented on e2e/cypress/integration/team/remove_team_icon_spec.js:
```diff
@@ -0,0 +1,106 @@
+        cy.fileUpload('[data-testid="uploadPicture"]', 'mattermost-icon.png');
+
+        // after uploading the picture the save button must be disabled
+        cy.get('[data-testid="saveSettingPicture"]').should('not.be.disabled').click();
```
This is not accurate. Currently, the save button is always `disable: false`.  To check enable/disable state of the button, I'd suggest to add ``disabled={disableSaveButtonFocus}`` (or change the variable to make it ``disabled={disableSaveButton}``) [here](https://github.com/mattermost/mattermost-webapp/pull/4611/files#diff-bf2e0895a3d11fd13d4f4a4f34feb18bR342), then check enable/disable before/after file upload.

VishalSwarnkar commented on e2e/cypress/integration/team/remove_team_icon_spec.js:
```diff
@@ -0,0 +1,106 @@
+        openTeamSettingsDialog();
+
+        // verify the settings picture button is visible to click
+        cy.get('[data-testid="inputSettingPictureButton"]').should('be.visible').click();
```
Hi @saturninoabril, thanks for the suggestion but as per @reflog review comments I have followed the https://docs.cypress.io/guides/references/best-practices.html and update the _ids_ to _data-testid_
Looking at the existing code base I can see the mix approach few of the specs has cy.findByTestId and few of them are using cy.get('[data-testid="xyz"]'). 
It would be really better if the approaches are consistent across, but please advise which approach is better?

saturninoabril replied:
It's not a contrary to his comment.  Since you're using `data-testid`, the convenience function is to get it by `cy.findByTestId`.  Those using ``cy.get('[data-testid="xyz"]')`` works too but since we have ``cy.findByTestId``, let's use it.
We can fix others in separate PR to make it consistent all throughout the codebase.

jasonblais commented:
> Hi @jasonblais the PR is now approved now but still, I can see the current status as "merge/blocked Expected — Waiting for status to be reported". Is there anything I have to do in regards to the PR?

@VishalSwarnkar Thanks! The PR actually undergoes two dev reviews and 1 QA review ([among the three reviews requested previously](https://github.com/mattermost/mattermost-webapp/pull/4611#event-2920941045)). You can learn more about our [code review process in our docs](https://developers.mattermost.com/contribute/getting-started/code-review/). Let me know if that's helpful.

As for the "merge/blocked Expected — Waiting for status to be reported" error, I'm not actually sure what that refers to, @saturninoabril would you know?

saturninoabril commented:
@jasonblais I'm also not sure.
@VishalSwarnkar Once you make changes per comments, please update your branch into master and let's see what'll happen with the "merge/block".

VishalSwarnkar commented:
Hi @saturninoabril as requested, I have incorporated the review comments and update the pull request. Kindly review

saturninoabril commented on components/setting_picture.jsx:
```diff
@@ -335,7 +338,8 @@ export default class SettingPicture extends Component {
                                 />
                             </button>
                             <button
-                                tabIndex={disableSaveButtonFocus ? '-1' : '0'}
```
Please revert this change which is for Accessibility.

VishalSwarnkar replied:
@saturninoabril reverted the changes and updated the pull request

VishalSwarnkar replied:
@saturninoabril reverted the changes
